FirebaseAuth/Tests/Unit/FIROAuthProviderTests.m

```diff
@@ -223,6 +261,8 @@ - (void)setUp {
   _mockRequestConfiguration = OCMClassMock([FIRAuthRequestConfiguration class]);
   OCMStub([_mockAuth requestConfiguration]).andReturn(_mockRequestConfiguration);
   OCMStub([_mockRequestConfiguration APIKey]).andReturn(kFakeAPIKey);
+  FIRKFakeAppCheck *fakeAppCheck = [[FIRKFakeAppCheck alloc] init];
```

Comment:
nit: Can you move these two lines inside `testGetCredentialWithUIDelegateWithAppCheckToken`? And then in the non-AppCheck test, you can verify no fragment was attached. (Same in FIRPhoneAuthProviderTests.m)

Reply:
Modify that in `FIROAuthProviderTest`, in `FIRPhoneAuthProviderTests`, the logic seems different, everything is tested together in the same test `verifyReCAPTCHAVerificationFlowWithUIDelegate`. So I will keep the original file for it. Love to modify it if I misunderstood the logic.

Reply:
Ah, in that case, can you add a separate test `verifyReCAPTCHAVerificationFlowWithUIDelegateAndAppCheckToken`? And then you can verify both the no fragment and yes fragment cases.